### PR TITLE
RAINCATCH-451 Update to use new simple store

### PR DESCRIPTION
**Pull request by JameelB**
## What
Updated server.js to add in an id field, required by the new raincatcher-simple-store module, to the workflow object upon creation.

## JIRA Ticket
https://issues.jboss.org/browse/RAINCATCH-451

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -6,4 +6,10 @@ var sync = require('fh-wfm-sync/lib/server')
 
 module.exports = function(mediator, app, mbaasApi) {
   sync.init(mediator, mbaasApi, config.datasetId, config.syncOptions);
+
+  //Adds an id field required by the new simple store to the workflow that will be created.
+  mediator.subscribe('wfm:cloud:workflows:create', function(workflowToCreate, ts) {
```
Comment (vchepeli) [inline lib/server.js]: whats `ts`?

**Comment by vchepeli** [comment]
LGTM

**Comment by vchepeli** [comment]
@JameelB, @nialldonnellyfh think do we need some unit tests for this?

File: lib/config.js [hunk 2]
```diff
@@ -8,5 +8,6 @@ module.exports = {
     "sync_frequency" : 5,
     "storage_strategy": "dom",
     "do_console_log": false
-  }
+  },
+  topicPrefix: 'wfm:cloud:'
```
Comment (nialldonnellyfh) [inline lib/config.js]: This should be `cloudTopicPrefix` as it is referencing the `wfm:cloud` topic name.

File: lib/server.js [hunk 3]
```diff
@@ -1,9 +1,48 @@
...
 
 
 module.exports = function(mediator, app, mbaasApi) {
   sync.init(mediator, mbaasApi, config.datasetId, config.syncOptions);
+
+  mediator.subscribe(config.topicPrefix + config.datasetId + ':create', function(workflowToCreate, mediatorTopicIdentifier) {
+    // Add an id field required by the new simple store module to the workflow object that will be created
+    workflowToCreate.id = shortid.generate();
+
+    mediator.request(config.topicPrefix + 'data:' + config.datasetId + ':create', workflowToCreate, {uid: workflowToCreate.id})
```
Comment (nialldonnellyfh) [inline lib/server.js]: It would be good to have a small function to generate topics to prevent errors in concatentation.

**Pull request merged**